### PR TITLE
introduced error and refactored to use one state

### PR DESCRIPTION
**Pull request by edewit**
fix for #67

### CODE REVIEW

File: src/app/wizard/deploy/deploy.component.ts [hunk 1]
```diff
@@ -64,8 +65,11 @@ export class DeployComponent {
                   }
                 }
 
-                this.done = this.statusMessages[this.statusMessages.length - 1].done;
-                if (this.done) this.webSocket.close();
+                let done = this.statusMessages[this.statusMessages.length - 1].done;
+                if (this.done) {
```
Comment (gastaldi) [inline src/app/wizard/deploy/deploy.component.ts]: Shouldn't the websocket be closed in case of error too?

Comment (edewit) [inline src/app/wizard/deploy/deploy.component.ts]: no because you can retry

**Comment by gastaldi** [comment]
But once you retry, isn't it a new Projectile + WebSocket connection ?




Best Regards,



George Gastaldi

Principal Software Engineer

Red Hat <https://www.redhat.com>

Remote

Joinville - Santa Catarina

ggastald@redhat.com    M: +55-47-99711-1000
<https://red.ht/sig>
TRIED. TESTED. TRUSTED. <https://redhat.com/trusted>
@redhatnews <https://twitter.com/redhatnews>   Red Hat
<https://www.linkedin.com/company/red-hat>   Red Hat
<https://www.facebook.com/RedHatInc>

File: src/app/wizard/deploy/deploy.component.ts (hunk 1 above)
Comment (edewit) [inline src/app/wizard/deploy/deploy.component.ts]: right, retry means new websocket connection, added the close on error

**Pull request merged**